File: python3/1800.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        s = nums[0]
        ms = nums[0]
        for i in range(1, len(nums)):
            s = nums[i] + (s if (nums[i - 1] < nums[i]) else 0)
            ms = max(ms, s)
        return ms

File: dart/1800.dart
class Solution {

    int max(int i, int j){
        return (i > j) ? i : j;
    }

    int maxAscendingSum(List<int> nums) {
        int s = nums[0], n = nums.length, ms = nums[0];
        for (int i = 1; i < n; i++) {
            s = nums[i] + ((nums[i - 1] < nums[i]) ? s : 0);
            ms = max(ms, s);
        }
        return ms;
    }
}
